### ISSUE TITLE
Multilabel support

### ISSUE THREAD

**Issue by adrianog** [sepandhaghighi/pycm#211]
#### Description
Add multilabel support

**Comment by sepandhaghighi** [comment]
We thought about this option, but it's not easy, most of `PyCM` parameters are not extendable to multilabel mode.
@alirezazolanvari what is your opinion ?

**Comment by adrianog** [comment]
These should be good:
Precision, recall and F1 score

And also:
Micro F1-score
Hamming loss
Jaccard index

Interesting link - micro and macro averaging - https://archive.is/fiF2C

**Comment by radiantone** [comment]
Does this ticket cover the case where you have something like this?

Actual
["LabelA","LabelB"]

Predicted
["LabelA","LabelC","LabelD"]

**Comment by sepandhaghighi** [comment]
> Does this ticket cover the case where you have something like this?
> 
> Actual
> ["LabelA","LabelB"]
> 
> Predicted
> ["LabelA","LabelC","LabelD"]

Hi @radiantone 
Yes, this issue talk about [multi-label classification](https://en.wikipedia.org/wiki/Multi-label_classification). 
For example : 
```pycon
>>> actual = [["LabelA","LabelB"],["LabelC","LabelB"],["LabelC"]]
>>> predict = [["LabelA","LabelB"],["LabelB"],["LabelA","LabelC","LabelD"]]
```
As I mentioned above, it's not easy, because most of `PyCM` parameters are not extendable to multi-label mode. We will work on it in the future.

**Comment by SoufianeDataFan** [comment]
Any alternatives that support Multi-labels classification report?

**Comment by sepandhaghighi** [comment]
> Any alternatives that support Multi-labels classification report?

You can use sickit-learn [multilabel confusion matrix](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.multilabel_confusion_matrix.html)
